pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -337,6 +342,67 @@ type GettableApiAlertingConfig struct {
 	Receivers []*GettableApiReceiver `yaml:"receivers,omitempty" json:"receivers,omitempty"`
 }
 
+func getMissingJSONTags(t reflect.Type, m map[string]string) {
+	if t.Kind() != reflect.Struct {
+		return
+	}
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		jsonTag := f.Tag.Get("json")
+		if jsonTag == "" {
+			if yamlTag, ok := t.Field(i).Tag.Lookup("yaml"); ok {
+				// TODO check for conflicts
+				m[f.Name] = strings.Split(yamlTag, ",")[0]
+			}
+		}
+		switch t.Field(i).Type.Kind() {
+		case reflect.Struct:
+			getMissingJSONTags(t.Field(i).Type, m)
```

Comment:
This concerns me a bit if users add label maps that have the same name as other fields (i.e. `{A,a}uthorization`), which would cause the user field's `json` representation to be overwritten.